src/crypto/equihash.cpp

```diff
@@ -39,6 +39,85 @@ int Equihash<N,K>::InitialiseState(eh_HashState& base_state)
                                                          personalization);
 }
 
+void ExpandArray(const unsigned char* in, size_t in_len,
+                 unsigned char* out, size_t out_len,
+                 size_t bit_len)
+{
+    assert(bit_len >= 8);
+    assert(8*sizeof(uint32_t) >= 7+bit_len);
+
+    size_t out_width { (bit_len+7)/8 };
+    assert(out_len == 8*out_width*in_len/bit_len);
```

Comment:
I think this assertion is subtly wrong; it should be
```
assert(out_len == (8*in_len/bit_len)*out_width);
```
It currently only works because ``bit_len`` happens to be a multiple of ``out_width``.